src/controllers/CheckoutController.php

```diff
@@ -53,15 +57,35 @@ public function actionCheckout(): Response
         $successUrl = $request->getValidatedBodyParam('successUrl');
         $cancelUrl = $request->getValidatedBodyParam('cancelUrl');
         $customer = $request->getBodyParam('customer');
+        // we're intentionally not passing any params from the form,
+        // so that you can't change what you get checked out with;
+        // you can pass params via EVENT_BEFORE_START_CHECKOUT_SESSION event
 
         if ($customer == 'false' || $customer == '0' || $customer === false || $customer === 0) {
             // if customer was explicitly set to something falsy,
             // go with false to prevent trying to find the currently logged in user further down the line
             $customer = false;
         }
 
+        $params = [];
+        $fields = $request->getBodyParam('fields');
+        if (!empty($fields)) {
+            // check the checkout mode - if it's subscription, proceed with creating a draft
+            $mode = $checkoutService->getCheckoutMode($lineItems);
+            if ($mode === StripeCheckoutSession::MODE_SUBSCRIPTION) {
+                // create an unpublished & unsaved draft subscription in Craft;
+                $subscription = new Subscription();
```

Comment:
Worth changing this to use the `createObject` method, just for future-proofing and DI?